outlines/models/transformers.py

```diff
@@ -0,0 +1,104 @@
+import math
+from typing import TYPE_CHECKING, List, Optional, Tuple, Union
+
+import numpy as np
+from numpy.typing import NDArray
+
+from outlines.models.tokenizer import Tokenizer
+
+if TYPE_CHECKING:
+    from transformers import PreTrainedModel, PreTrainedTokenizer
+
+
+__all__ = ["transformers"]
+
+
+class Transformers:
```

Comment:
Should we add a model protocol, or wait until we have several model implementations?